tests/test_recipient_csv.py

```diff
@@ -422,6 +422,21 @@ def test_overly_big_list_stops_processing_rows_beyond_max(mocker):
     ) == 10
 
 
+def test_rows_processing_timeout():
+    recipients = RecipientCSV(
+        # Keep the number of rows small to avoid DoSing other tests
+        'phone_number,07900900900\n' * 1000,
+        template=_sample_template('sms'),
+        # Use "0" to avoid flakey behaviour based on the no. of rows
+        processing_timeout=0,
+    )
+
+    with pytest.raises(RuntimeError) as excinfo:
+        assert not recipients.rows
```

Comment:
Good point. In theory it's fine because `recipients.rows` should never run to completion due to the timeout, so the `assert not` would correctly fail if it did. However, you're right that this is covered by `pytest.raises`.

I've removed it.